File: Leetcode/136. Single Number/solution.rb
def single_number(nums)
  hashNums = {}

  nums.each do |num|
    if hashNums[num]
      hashNums[num] += 1
    else
      hashNums[num] = 1
    end
  end

  hashNums.key(1)
end

print single_number([4,1,2,1,2])

File: Leetcode/136. Single Number/solution.js
const singleNumber = (nums) => {
  const objNums = {}
  
  for(el of nums){
    if(objNums[el]){
      objNums[el] += 1
    } else {
      objNums[el] = 1
    }
  }

  return Object.keys(objNums).find(key => objNums[key] === 1)
}

console.log(singleNumber([4,1,2,1,2]))